crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -86,3 +217,161 @@ impl From<TradeError> for PriceEstimationError {
         }
     }
 }
+
+/// Output of `Trader::settle` smart contract call.
+#[derive(Debug, Eq, PartialEq)]
+struct SettleOutput {
+    gas_used: U256,
+    trader_balances: Vec<I256>,
+    settlement_balances: Vec<I256>,
+}
+
+impl SettleOutput {
+    fn decode(output: &[u8]) -> Result<Self> {
+        let function = Trader::raw_contract().abi.function("settle").unwrap();
+        let tokens = function.decode_output(output)?;
+        let (gas_used, trader_balances, settlement_balances) =
+            Tokenize::from_token(Token::Tuple(tokens))?;
+        Ok(Self {
+            gas_used,
+            trader_balances,
+            settlement_balances,
+        })
+    }
+
+    fn trader_amounts(&self) -> Option<(U256, U256)> {
+        Some((
+            self.trader_balances.get(0)?.wrapping_abs().into_raw(),
```

Comment:
We use `wrapping_abs().into_raw()`a bunch here. Is that correct instead of an erroring conversion?

Reply:
It should never error. Ideally, `I256` would have a `unsigned_abs()`, but it doesn't. `.wrapping_abs().into_raw()` should be equivalent.